src/databases/functions/users/user-set-karma-score.js

```diff
@@ -1,10 +1,34 @@
 const {User} = require('../../models');
+const {QueryTypes} = require('sequelize');
+const db = require('../../models/');
 
 module.exports = async (userId, combined_user_score, karma_score) => {
   if (!userId) return null;
   try {
+    let total_follower = 0;
+    const queryTotalFollower = `
+      SELECT user_id_followed, COUNT(user_id_follower) AS follower_count
+      FROM user_follow_user
+      WHERE
+      is_anonymous = false
+      AND user_id_followed = :userId
+      GROUP BY user_id_followed
+    `;
+
+    const resultTotalFollower = await db.sequelize.query(queryTotalFollower, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+
+    if (queryTotalFollower) {
+      total_follower = resultTotalFollower[0]?.follower_count;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a logical error here. You are checking if `queryTotalFollower` is truthy, but `queryTotalFollower` is a string and it's always truthy. I believe you meant to check if `resultTotalFollower` has any results.
```diff
-    if (queryTotalFollower) {
+    if (resultTotalFollower.length > 0) {
       total_follower = resultTotalFollower[0]?.follower_count;
     }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Please see this coderabbit recommendation

Reply:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

@fajarism, you're absolutely correct and I appreciate your attention to detail. The check should indeed be for `resultTotalFollower` instead of `queryTotalFollower`. Here's the corrected code snippet:

```diff
-    if (queryTotalFollower) {
+    if (resultTotalFollower.length > 0) {
       total_follower = resultTotalFollower[0]?.follower_count;
     }
```

This change ensures that we only attempt to access `follower_count` if there are results from the query. Thank you for pointing out this logical error.

<!-- This is an auto-generated reply by OSS CodeRabbit -->